experimenter/experimenter/experiments/migrations/0266_nimbusexperiment_results_pairwise_data.py

```diff
@@ -0,0 +1,181 @@
+# Generated by Django 3.2.23 on 2024-01-24 21:10
+
+from copy import deepcopy
+from django.db import migrations, models
+
+
+def update_results_data_schema(apps, schema_editor):
+    NimbusExperiment = apps.get_model("experiments", "NimbusExperiment")
+
+    for experiment in NimbusExperiment.objects.all():
+        branches = [b.slug for b in experiment.branches.all()]
+        reference_branch = experiment.reference_branch
+        data = experiment.results_data
+        if data is not None and "v3" not in data and "v2" in data:
+            data["v3"] = data["v2"]
+            """
+            data for v3 looks like this (with changes vs v2 noted):
+            [window]: {
+                [analysis_basis]: {
+                    [segment]: {
+                        [branch]: {
+                            "branch_data": {
+                                [metrics_group]: {
+                                    [metric_name]: {
+                                        "absolute": {
+                                            "all": [...],
+                                            "first": {...}
+                                        },
+                                        "difference": {
+                                            // "all" and "first" moved from here into [comparison_branch]
+                                            [comparison_branch]: { // this level is new
+                                                "all": [...],
+                                                "first": {...},
+                                            },
+                                        },
+                                        "relative_uplift": {
+                                            // "all" and "first" moved from here into [comparison_branch]
+                                            [comparison_branch]: { // this level is new
+                                                "all": [...],
+                                                "first": {...},
+                                            },
+                                        },
+                                        "significance": {
+                                            // optional daily/weekly/overall moved from here into [comparison_branch]
+                                            [comparison_branch]: { // this level is new
+                                                "weekly": {},
+                                                "overall: {}
+                                            }
+                                        },
+                                        "percent": 0.0
+            """
+
+            comparison_default = {"first": {}, "all": []}
+            significance_default = {"weekly": {}, "overall": []}
+
+            for key, value in data["v3"].items():
+                if value is not None and key in ["weekly", "overall"]:
+                    for basis, basis_data in value.items():
```

Comment:
Might I suggest inverting this condition and adding `continue` so we don't nest so deeply?

```suggestion
                if value is None or key in ["weekly", "overall"]:
                    continue
                    
                for basis, basis_data in value.items():
```